llvm/lib/CodeGen/GCMetadata.cpp

```diff
@@ -84,6 +142,36 @@ void GCModuleInfo::clear() {
 
 // -----------------------------------------------------------------------------
 
+PreservedAnalyses GCInfoPrinterPass::run(Function &F,
+                                         FunctionAnalysisManager &FAM) {
+  if (F.isDeclaration() || !F.hasGC())
+    return PreservedAnalyses::all();
+
+  auto &Info = FAM.getResult<GCFunctionAnalysis>(F);
```

Comment:
A lot of this still seems to be duplicated with the legacy printer pass? Is there a way to share the implementation between them? Both seem to just be iterating over a `GCFunctionInfo` class.